### ISSUE TITLE
@typescript-eslint/no-unused-expressions for $: when upgrading to eslint 9.7.0

### ISSUE THREAD

**Issue by jackie35er** [sveltejs/eslint-plugin-svelte#934]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.7.0

### What version of `eslint-plugin-svelte` are you using?

2.46.0

### What did you do?

<details>
<summary>Configuration</summary>

```
import js from '@eslint/js';
import svelte from 'eslint-plugin-svelte';
import globals from 'globals';
import ts from 'typescript-eslint';

export default ts.config(
  js.configs.recommended,
  ...ts.configs.recommended,
  ...svelte.configs["flat/recommended"],
  {
    languageOptions: {
	  globals: {
	    ...globals.browser,
	    ...globals.node
	  }
	}
  },
  {
    files: ["**/*.svelte"],

    languageOptions: {
	  parserOptions: {
	    parser: ts.parser
	  }
	}
  },
  {
    ignores: ["build/", ".svelte-kit/", "dist/"]
  }
);

```
</details>

```svelte
<script lang="ts">

    let value: string;

    $: value, myFunction();

    function myFunction() {
        console.log("Hello World!")
    }

</script>

<input bind:value>
```

### What did you expect to happen?

Nothing

### What actually happened?

` 5:8  error  Expected an assignment or function call and instead saw an expression  @typescript-eslint/no-unused-expressions`

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/jackie35er/svelte-eslint-bug-example/tree/master

**Comment by ota-meshi** [comment]
In your example, I don't think you actually use that expression.
Could you explain in more detail what the problem is?

**Comment by jackie35er** [comment]
The example above is perfectly valid svelte code. Combining the $ operater with the , operator leads to myFunction() getting called everytime value changes. This worked before upgrading from eslint 8.57.1.

**Comment by jackie35er** [comment]
see https://svelte.dev/playground/bafddb4c8cdd4c9c8187ca0a06e5f72f?version=5.2.10

**Comment by ota-meshi** [comment]
> The example above is perfectly valid svelte code. 

I believe all code that the no-unused-expressions rule reports is valid, and if you don't like the rule you can turn it off.

**Comment by jackie35er** [comment]
In my opinion this is a false positive. The expression is used and removing it would change the behaviour of the code changes when removing it.

**Comment by baseballyama** [comment]
We can easily address this issue by passing dummy arguments. Generally, cases where a tracked value changes but is not used at all in the process are rare. Given that this can be resolved simply by using dummy arguments, we’ve decided to hold off on making any code changes for now and monitor the situation.

If similar requests come frequently, we will reopen the issue, but for now, we will close this issue.

```ts
$: {
  myFunction(value);
}
```
